protocols/raft/src/main/java/io/atomix/protocols/raft/impl/RaftContext.java

```diff
@@ -201,10 +202,25 @@ public RaftContext(
     registerHandlers(protocol);
 
     raftRoleMetrics = new RaftRoleMetrics(name);
+    started = true;
   }
 
   private void onUncaughtException(Throwable error) {
-    log.error("An uncaught exception occurred", error);
+    log.error("An uncaught exception occurred, transition to inactive role", error);
+    try {
+      // to prevent further operations submitted to the threadcontext to execute
+      transition(Role.INACTIVE);
```

Comment:
If already inactive, we could maybe shut down? Very unlikely that we get an uncaught error if inactive, but who knows.

Reply:
An exception in Inactive role, if happens at all, should not lead to an inconsistent state. So we could just leave it as it is. Closing Raft should be done only if there is not other way to prevent inconsistent state. Other brokers are still trying to communicate to this node. So we should leave Raft running in inactive if possible.

Reply:
So we could simply skip if we're already inactive?

Reply:
It will be skipped in `transition`. It checks the current role before transitioning.